### PR TITLE
Use MapAxis in gammapy.irf consistently

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request does a major refactoring by using `MapAxis` objects in the IRF class init methods consistently. In addition it introduces the following (internal) naming convention for IRF axes:

- ``energy``     -   Reconstructed energy axis
- ``energy_true`` -  True energy axis
- ``offset`` - Field of view offset from center
- ``fov_lon`` - Field of view	longitude
- ``fov_lat`` - Field of view latitude
- ``migra`` - Energy migration
- ``rad``  - Offset angle from source position

I'm happy to re-discuss names later, for now its just important that they names are used consistently.

### CODE REVIEW

File: gammapy/makers/tests/test_map.py [hunk 1]
```diff
@@ -216,135 +216,7 @@ def test_make_meta_table(observations):
     assert_allclose(map_dataset_meta_table["OBS_ID"], 110380)
 
 
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
```
Comment (adonath) [inline gammapy/makers/tests/test_map.py:219]: This test was completely duplicated, that's why I removed it...

**Review by AtreyeeS — APPROVED** [review]
Thanks @adonath ! This is a very welcome cleanup!

File: docs/irf/index.rst [hunk 2]
```diff
@@ -26,6 +26,23 @@ EDISP and PSF for a given dataset.
...
+================= ===================================
+Variable          Definition
+================= ===================================
+``energy``        Reconstructed energy axis
+``energy_true``   True energy axis
+``offset``        Field of view offset from center
+``fov_lon``       Field of view	longitude
+``fov_lat``       Field of view latitude
+``migra``         Energy migration
+``rad``        	  Offset angle from source position
```
Comment (registerrier) [inline docs/irf/index.rst:43]: Maybe explicitly mention PSF in the `rad` definition and energy dispersion in the `migra` definition.

Comment (registerrier) [inline docs/irf/index.rst:43]: Maybe define `migra` as the correct ratio and mention energy dispersion.
Also mention PSF in `rad`definition

**Review by registerrier — APPROVED** [review]
Thanks @adonath ! A needed change.

One small suggestion for the doc inline.

**Comment by adonath** [comment]
Thanks @AtreyeeS and @registerrier! @registerrier I'll address your comments in a follow up PR.

**Pull request merged**